docs/oauth.md

```diff
@@ -0,0 +1,169 @@
+# OAuth2 API
+
+OAuth2 is a protocol that allows applications to use Webmaker accounts without needing to store a username and password.
+
+Before you can use OAuth2 you'll need to register your application.
+
+## Registering your application
+
+Webmaker requires the following information to register your application:
+
+* **Name** - The friendly name of your application e.g. "Buckley's Bees".
+* **Description** - A description of your application e.g. "Buckley's Bees sells the best honey in Ontario".
+* **Image** - An image that represents your application.
+* **Homepage URL** - The full URL to your application homepage e.g. "https://buckleysbees.ca".
+* **Redirect URL** - Your application's OAuth2 callback URL e.g. "https://buckleysbees.ca/oauth2/callback".
+
+Once you've registered your application you'll recieve your application credentials:
+
+* **Client ID** - A string identifying your application. This value is not secret and can be shown publicly e.g. `wm_id_BTQNPABtUqqApaDrcsDa`.
+* **Client Secret** - A string. This value is secret and must not be shared with anyone e.g. `wm_secret_iHmAWhvCwBYnuE6aZHcArBEPohanKmmWr8LcyBBPZYRkAxdmui`.
+
+You can register your application by emailing [jbuck](mailto:jon@mozillafoundation.org). Self-serve application registration is coming shortly.
+
+## Web Application Flow
+
+The web application flow is ideal for services that have a server-side component to securely store the client secret and user's access tokens.
+
+### 1. Redirect users to request access
+
+To start the web application flow you need to redirect your users browser to Webmaker.
+
+`GET https://id.webmaker.org/login/oauth/authorize`
+
+**Parameters**
+
+<table>
+  <tr>
+  	<th>Name</th>
+  	<th>Description</th>
+  </tr>
+  <tr>
+  	<td>client_id</td>
+  	<td>The client ID you received when you registered your application.</td>
+  </tr>
+  <tr>
+  	<td>response_type</td>
+  	<td>The response_type must be set to "code".</td>
+  </tr>
+  <tr>
+  	<td>scopes</td>
+  	<td>A space separated list of scopes.</td>
+  </tr>
+  <tr>
+  	<td>state</td>
+  	<td>An unguessable random string. It is used to protect against cross-site request forgery attacks.</td>
```

Comment:
s/unguessable/secure/ ? Also, probably good to mention this should be random every time, because it is used in a handshake round trip, with the value going into the request, the oauth2 doing things and sending it back during callback, so that the requester can verify that the answer it got comes with the same "state" that it sent out?